crates/telio-traversal/src/endpoint_providers/upnp.rs

```diff
@@ -653,7 +653,7 @@ impl<Wg: WireGuard, I: UpnpEpCommands, E: Backoff> Runtime for State<Wg, I, E> {
     {
         pin!(updated);
 
-        if self.is_endpoint_provider_paused {
+        if !self.igd_gw.lease_needs_renew() && self.is_endpoint_provider_paused {
```

Comment:
I think we may have a bit of an issue with pausing and UPnP module. (And we had it for as long as UPnP and pausing module existed, but I guess it would be nice to fix it.

As pausing is done on the assumption, that endpoint will stay "active" without any actions, which is true for local endpoints and STUN endpoints (as long as traffic is flowing), but it is *not* true for UPnP, as the gateway have every right to cut off the traffic flow once lease expires. 

In our case, we will check whether there are no pending renews required and we will check if module is paused -> and will go to indefinite sleep. During which the lease may expire and gateway could cutoff communication. 

So I wonder if pausing even makes sense for UPnP at all :thinking:.

Reply:
Yeah, I was also worried about that but this is not exactly how it will behave. Every 5 seconds we will run periodic wg consolidation. This will pause the endpoint providers even if they are already pause which other branch of the `select!` and returns. Which means that we will notice the need for renew with 5s granularity.

Reply:
Hmm, this requirement to call "pause()" every once in a while seems like really non-obvious requirement. I know that this is not directly related to your task, but are there any ways to make this requirement more explicit?

Reply:
We could ignore the question of how to enforce pausing, by replacing this [line](https://github.com/NordSecurity/libtelio/blob/b92ad778b7a7eb3092dfc63bdc255725081cbe25/crates/telio-traversal/src/endpoint_providers/upnp.rs#L659) with a await for sleep(5s)

Reply:
yeah, that could work.

On the other hand, `lease_needs_renew()` does not when exactly renew will be needed, so maybe it is worth `::sleep_until()`'ing instead of polling?

Reply:
Good point. I've updated the code so that we sleep until the time when renew is needed.